### PR TITLE
MINIFICPP-2312 Reduce shared_ptr usage for controller services

### PR DESCRIPTION
**Pull request by fgerlits**
I have removed some `shared_ptr` usage, but more can be done later.

The main problem is that the `ControllerServiceNodeMap` container is shared between the `FlowConfiguration`, the `ProcessGroup` and the `ProcessContext`.  It should probably be created in the `FlowConfiguration`, moved to the `ProcessGroup`, and the `ProcessContext` would only have a pointer to it.

I have renamed `ControllerServiceMap` to `ControllerServiceNodeMap` (because it stores `ControllerServiceNode`s), but git shows it as a removed file plus a new file.

https://issues.apache.org/jira/browse/MINIFICPP-2312

---

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [x] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Review by lordgamez — APPROVED** [review]

**Review by adamdebreceni — APPROVED** [review]

File: extensions/http-curl/tests/ControllerServiceIntegrationTests.cpp [hunk 1]
```diff
@@ -90,31 +89,30 @@ int main(int argc, char **argv) {
...
     controller->load();
     controller->start();
-    ssl_client_cont = controller->getControllerServiceNode("SSLClientServiceTest");
-    ssl_client_cont->enable();
-    assert(ssl_client_cont != nullptr);
-    assert(ssl_client_cont->getControllerServiceImplementation() != nullptr);
-    ssl_client = std::static_pointer_cast<minifi::controllers::SSLContextService>(ssl_client_cont->getControllerServiceImplementation());
+    auto* ssl_client_node = controller->getControllerServiceNode("SSLClientServiceTest");
+    ssl_client_node->enable();
+    assert(ssl_client_node != nullptr);
```
Comment (szaszm) [inline extensions/http-curl/tests/ControllerServiceIntegrationTests.cpp:109]: Can we switch these lines?

**Review by szaszm — APPROVED** [review]
Small unrelated comment, but LGTM either way

File: extensions/http-curl/tests/ControllerServiceIntegrationTests.cpp (hunk 1 above)
Comment (szaszm) [reply extensions/http-curl/tests/ControllerServiceIntegrationTests.cpp]: I pushed a commit with this: [9ff2075](https://github.com/apache/nifi-minifi-cpp/pull/1771/commits/9ff207553e04f6ab06f17699debdb8d3eab1c3c5)
Revert it if you disagree

Comment (fgerlits) [reply extensions/http-curl/tests/ControllerServiceIntegrationTests.cpp]: yes, good catch

**Pull request closed without merging**